src/main/java/net/neoforged/neoforge/common/data/internal/NeoForgeItemTagsProvider.java

```diff
@@ -101,7 +101,7 @@ public void addTags(HolderLookup.Provider lookupProvider) {
         copy(Tags.Blocks.FENCES_NETHER_BRICK, Tags.Items.FENCES_NETHER_BRICK);
         copy(Tags.Blocks.FENCES_WOODEN, Tags.Items.FENCES_WOODEN);
         tag(Tags.Items.FERTILIZERS).add(Items.BONE_MEAL);
-        tag(Tags.Items.FOODS_FRUIT).add(Items.APPLE, Items.GOLDEN_APPLE, Items.ENCHANTED_GOLDEN_APPLE);
+        tag(Tags.Items.FOODS_FRUIT).add(Items.APPLE, Items.GOLDEN_APPLE, Items.ENCHANTED_GOLDEN_APPLE).add(Items.MELON);
         tag(Tags.Items.FOODS_VEGETABLE).add(Items.CARROT, Items.GOLDEN_CARROT, Items.POTATO, Items.MELON_SLICE, Items.BEETROOT);
```

Comment:
This retains the addition of the melon slice item in the vegetables tag, which was removed in the corresponding Fabric PR: https://github.com/FabricMC/fabric/pull/3957/files#diff-bd8d1f7f942977cc89b7d65685dd0327a87c3656352c817981c821fd142fb269L227.